features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIClient.js

```diff
@@ -134,14 +134,44 @@ class APIClient {
 
     _getRequestInterceptor() {
         return (request) => {
-            AuthManager.refreshTokenOnExpire(request, this.environment);
+            // todo: sync token checking logic
+            const existingUser = AuthManager.getUser(this.environment.label);
+            if (!existingUser) {
+                console.log('User not found. Token refreshing failed.');
+                return request;
+            }
+            const existingToken = AuthManager.getUser(this.environment.label).getPartialToken();
+            const refToken = AuthManager.getUser(this.environment.label).getRefreshPartialToken();
+            if (existingToken && (existingToken !== 'undefined')) {
```

Comment:
Is there any possibility of returning `'undefined'` string from `getPartialToken()` method ?

otherwise `&& (existingToken !== 'undefined')` part is redundent here

Reply:
only if the value of the cookie gets empty. Just to avoid such an error scenario.

Reply:
Yes I think that could happen only [if this](https://github.com/wso2/carbon-apimgt/blob/master/features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/Utils.js#L39) `pair[1]` get undefined.

So , IMO it's better we handle it there and return null , if the value is `'undefined'` (note: string)